decidim-core/app/jobs/decidim/delete_inactive_participants_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A job to delete inactive participants and notify them
+  class DeleteInactiveParticipantsJob < ApplicationJob
+    queue_as :delete_inactive_participants
+
+    INITIAL_WARNING_PERIOD_DAYS = 30
+    FINAL_REMINDER_PERIOD_DAYS = 7
+
+    def perform(organization)
+      query = InactiveUsersQuery.new(
+        organization,
+        inactivity_period,
+        INITIAL_WARNING_PERIOD_DAYS,
+        FINAL_REMINDER_PERIOD_DAYS
+      )
+
+      unmark_active_users(query.users_to_unmark_for_deletion)
+      mark_users_for_deletion(query.users_to_mark_for_deletion)
+      send_reminder_notifications(query.users_to_send_reminder)
+      remove_inactive_users(query.users_to_remove)
```

Comment:
Alternatively, we could just use the proposed scopes in the User model:

```suggestion
  send_first_warning_notifications(organization.users.first_warning_inactive_users)
  send_last_warning_notifications(organization.users.last_warning_inactive_users)
  remove_inactive_users(organization.users.inactive_users)
```